src/RuleAction.h

```diff
@@ -24,32 +31,31 @@ class RuleAction {
 class RuleActionEvent : public RuleAction {
 public:
     explicit RuleActionEvent(const char* arg_msg);
-    ~RuleActionEvent() override { delete[] msg; }
+    explicit RuleActionEvent(const char* arg_msg, const char* event_name);
 
     void DoAction(const Rule* parent, RuleEndpointState* state, const u_char* data, int len) override;
 
     void PrintDebug() override;
 
 private:
-    const char* msg;
+    StringValPtr msg;
+    EventHandlerPtr handler;
 };
 
 class RuleActionMIME : public RuleAction {
 public:
     explicit RuleActionMIME(const char* arg_mime, int arg_strength = 0);
 
-    ~RuleActionMIME() override { delete[] mime; }
-
     void DoAction(const Rule* parent, RuleEndpointState* state, const u_char* data, int len) override {}
 
     void PrintDebug() override;
 
-    std::string GetMIME() const { return mime; }
+    const std::string& GetMIME() const { return mime; }
 
     int GetStrength() const { return strength; }
 
 private:
-    const char* mime;
+    std::string mime;
```

Comment:
is improving `mime` here needed for this change, or just a makes-sense-to-tidy-while-we're-in-there tweak?

Reply:
Yep, separate commit and has the "makes sense part" :-)